File: C++/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max = nums[0];
        int tmp = 0;
        for (int i = 0; i < nums.size(); i++){
            tmp += nums[i];
            if (tmp > max)
                max = tmp;
            if(tmp < 0)
                tmp = 0;
        }
        return max;
    }
};

File: Python/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if nums is None or len(nums) == 0:
            return None
        ans = nums[0]
        tmp = 0
        for num in nums:
            tmp += num
            ans = max(tmp, ans)
            if tmp < 0:
                tmp = 0
        return ans
